dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/WorkspaceItemRestRepository.java

```diff
@@ -0,0 +1,315 @@
+		Bitstream source = null;
+		BitstreamFormat bf = null;
+
+		Context context = obtainContext();
+		WorkspaceItem wsi = wis.find(context, id);
+		Item item = wsi.getItem();
+		// do we already have a bundle?
+		List<Bundle> bundles = itemService.getBundles(item, Constants.CONTENT_BUNDLE_NAME);
+
+		try {
+			InputStream inputStream = new BufferedInputStream(file.getInputStream());
+			if (bundles.size() < 1) {
+				// set bundle's name to ORIGINAL
+				source = itemService.createSingleBitstream(context, inputStream, item, Constants.CONTENT_BUNDLE_NAME);
+			} else {
+				// we have a bundle already, just add bitstream
+				source = bitstreamService.create(context, bundles.get(0), inputStream);
+			}
+		} catch (Exception e) {
+			log.error(e.getMessage(), e);
+			result = new UploadBitstreamRest();
+			result.setMessage(e.getMessage());
+			result.setStatus(false);
+			return result;
+		}
+
+		source.setName(context, file.getOriginalFilename());
+		// TODO how retrieve this information?
+		source.setSource(context, extraField);
+
+		// Identify the format
+		bf = bitstreamFormatService.guessFormat(context, source);
+		source.setFormat(context, bf);
+
+		// Update to DB
+		bitstreamService.update(context, source);
+		itemService.update(context, item);
+		context.commit();
+
+		result = submissionService.buildUploadBitstream(configurationService, source);
+		result.setStatus(true);
+		return result;
+	}
+
+	@Override
+	public void patch(Context context, HttpServletRequest request, String apiCategory, String model, Integer id, Patch patch) throws SQLException, AuthorizeException {
+		List<Operation> operations = patch.getOperations();
+		WorkspaceItemRest wsi = findOne(id);
+		WorkspaceItem source = wis.find(context, id);
+		for(Operation op : operations) {
+			//the value in the position 0 is a null value
+			String[] path = op.getPath().substring(1).split("/",3);
+			if(OPERATION_PATH_SECTIONS.equals(path[0])) {
+				String section = path[1];
+				evaluatePatch(context, request, source, wsi, section, op);
+			}
+			else {
+				throw new PatchBadRequestException("Patch path operation need to starts with '" + OPERATION_PATH_SECTIONS + "'");
+			}
+		}
+		wis.update(context, source);
+	}
+
+	private void evaluatePatch(Context context, HttpServletRequest request, WorkspaceItem source, WorkspaceItemRest wsi, String section, Operation op) {
+		SubmissionConfig submissionConfig = submissionConfigReader.getSubmissionConfigByName(wsi.getSubmissionDefinition().getName());
+		for(int stepNum = 0; stepNum<submissionConfig.getNumberOfSteps(); stepNum++) {
+			
+			SubmissionStepConfig stepConfig = submissionConfig.getStep(stepNum);
+			
+			if (section.equals(stepConfig.getId())) {
+				/*
+				 * First, load the step processing class (using the current
+				 * class loader)
+				 */
+				ClassLoader loader = this.getClass().getClassLoader();
+				Class stepClass;
+				try {
+					stepClass = loader.loadClass(stepConfig.getProcessingClassName());
```

Comment:
This looks like the third copy of the code for loading the processing steps plugins

Reply:
As above.